From 6519c5195084b7326aa8a26584bb2e2b4d82f19c Mon Sep 17 00:00:00 2001
From: He Sheng <hesheng@wxiat.com>
Date: Fri, 6 May 2022 10:05:54 +0800
Subject: [PATCH] sw64: move ucontext.h to uapi
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 6519c5195084b7326aa8a26584bb2e2b4d82f19c
Modified-by-SEL: No


Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I56OLG

--------------------------------

It's an important user interface which should be put in uapi.

Signed-off-by: He Sheng <hesheng@wxiat.com>

Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/include/asm/ucontext.h      | 14 --------------
 arch/sw_64/include/uapi/asm/ucontext.h | 14 ++++++++++++++
 2 files changed, 14 insertions(+), 14 deletions(-)
 delete mode 100644 arch/sw_64/include/asm/ucontext.h
 create mode 100644 arch/sw_64/include/uapi/asm/ucontext.h

diff --git a/arch/sw_64/include/asm/ucontext.h b/arch/sw_64/include/asm/ucontext.h
deleted file mode 100644
index d40eebe988ef..000000000000
--- a/arch/sw_64/include/asm/ucontext.h
+++ /dev/null
@@ -1,14 +0,0 @@
-/* SPDX-License-Identifier: GPL-2.0 */
-#ifndef _ASM_SW64_UCONTEXT_H
-#define _ASM_SW64_UCONTEXT_H
-
-struct ucontext {
-	unsigned long		uc_flags;
-	struct ucontext		*uc_link;
-	old_sigset_t		uc_old_sigmask;
-	stack_t			uc_stack;
-	struct sigcontext	uc_mcontext;
-	sigset_t		uc_sigmask;	/* mask last for extensibility */
-};
-
-#endif /* _ASM_SW64_UCONTEXT_H */
diff --git a/arch/sw_64/include/uapi/asm/ucontext.h b/arch/sw_64/include/uapi/asm/ucontext.h
new file mode 100644
index 000000000000..c5d6e24e3e5f
--- /dev/null
+++ b/arch/sw_64/include/uapi/asm/ucontext.h
@@ -0,0 +1,14 @@
+/* SPDX-License-Identifier: GPL-2.0 WITH Linux-syscall-note */
+#ifndef _UAPI_ASM_SW64_UCONTEXT_H
+#define _UAPI_ASM_SW64_UCONTEXT_H
+
+struct ucontext {
+	unsigned long		uc_flags;
+	struct ucontext		*uc_link;
+	old_sigset_t		uc_old_sigmask;
+	stack_t			uc_stack;
+	struct sigcontext	uc_mcontext;
+	sigset_t		uc_sigmask;	/* mask last for extensibility */
+};
+
+#endif /* _UAPI_ASM_SW64_UCONTEXT_H */
-- 
2.33.0

